gaps: make workspace gap assignments order-independent

stapelberg opened the pull request:
This commit moves subtracting the global gaps from the workspace gaps: previously, this calculation was done while parsing the configuration (order dependent), now it’s done at workspace assignment evaluation time.

related to https://github.com/i3/i3/issues/3724

fixes https://github.com/i3/i3/issues/5253

stapelberg committed:
gaps: make workspace gap assignments order-independent

This commit moves subtracting the global gaps from the workspace gaps:
previously, this calculation was done while parsing the configuration
(order dependent), now it’s done at workspace assignment evaluation time.

related to https://github.com/i3/i3/issues/3724

fixes https://github.com/i3/i3/issues/5253

Airblader reviewed (approved):
I can't help but point out that this is technically a breaking change in behavior. Sure it wasn't intended and discouraged to have them in the wrong order, but people may have done it anyway and adjusted the values to account for it. Probably still fine, but perhaps we should make a note?

The thing we definitely should do is remove the warning from the user guide that the configuration is order dependent.

stapelberg commented:
> I can't help but point out that this is technically a breaking change in behavior. Sure it wasn't intended and discouraged to have them in the wrong order, but people may have done it anyway and adjusted the values to account for it. Probably still fine, but perhaps we should make a note?

Agreed, added a release note. 

Also removed the note from the userguide.
